labelbox/data/generator.py

```diff
@@ -64,19 +77,22 @@ def fill_queue(self):
                     raise ValueError("Unexpected None")
                 self.queue.put(value)
         except Exception as e:
-            logger.warning("Unexpected exception while filling the queue. %r",
-                           e)
-        finally:
-            self.queue.put(None)
+            self.queue.put(
+                ValueError("Unexpected exception while filling queue. %r", e))
 
     def __iter__(self):
         return self
 
     def __next__(self) -> Any:
-        if self.done:
+        if self.done or self.queue.empty():
             raise StopIteration
         value = self.queue.get()
+        if isinstance(value, ValueError):
+            raise value
```

Comment:
this solution fixes both multithread and non-multithread issue right. So this covers both tickets?

Reply:
I believe so!